analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ClassShouldNotBeEmpty.Inheritance.cs

```diff
@@ -18,6 +19,9 @@ abstract class AbstractBaseWithoutAbstractMethods
 class NoImplementation: AbstractBaseWithAbstractMethods { }          // Error - abstract methods should be implemented
 class DefaultImplementation: AbstractBaseWithoutAbstractMethods { }  // Compliant - the class will use the default implementation of DefaultMethod
 
-class EmptyPageModel: PageModel { }                                  // Compliant - an empty PageModel can be fully functional, the C# code can be in the cshtml file
 class CustomException: Exception { }                                 // Compliant - empty exception classes are allowed, the name of the class already provides information
+class CustomAttribute: Attribute { }                                 // Compliant - empty attribute classes are allowed, the name of the class already provides information
+
+class EmptyPageModel: PageModel { }                                  // Compliant - an empty PageModel can be fully functional, the C# code can be in the cshtml file
+class CustomActionResult: ActionResult { }                           // Compliant - an empty action result can still provide information by its name
```

Comment:
also IActionResult and a custom type implementing IActionResult (FooActionResult)?

```
class BarActionResult : IActionResult {}
class FooActionResult: BarActionResult { }
```

Reply:
I'd get a compilation error for not implementing the interface methods. ASP.NET Core has several classes implementing the interface (like `ActionResult`). It's easier to check if the base type implements `IActionResult`, then to look for all the base types in ASP.NET Core which implement `IActionResult` and check if the analyzed class derives from that.